src/heap/heap.ts

```diff
@@ -0,0 +1,123 @@
+interface IHeap {
+  insert: (key: number) => void;
+  delete: () => number | null;
+  peek: () => number | null;
+}
+
+export enum HeapType {
```

Comment:
Look how enum is transpiled into JS 
<img width="1352" alt="Screenshot 2023-01-12 at 11 39 47" src="https://user-images.githubusercontent.com/14075402/212032369-fc0e39da-9215-47b6-be71-c1ecd6a4e6ea.png">

Try to never use them and always convince people to avoid them.
Use just const instead:
```typescript
const HEAP_TYPE = {
MAX = 'MAX',
MIN = 'MIN'
}
```
or class with the static fields
```typescript
class HeapType {
  public static readonly MAX = 'MAX';
  public static readonly MIN = 'MIN';
}
```

Also you can use as a values numbers `0` and `1` - they'll take less space in the memory)

Reply:
Thanks for the explanation and helpful advice 👍